### DISCUSSION TITLE
Support private certificate authority

### DISCUSSION THREAD

**Discussion by gaeljw** [aquasecurity/trivy#4194] [Q&A] [answered]
When using Trivy against a private image registry which HTTPS certificate is signed by a private authority, the error described in https://github.com/aquasecurity/trivy/issues/757 is raised:

```
Get "https://registry.private/v2/": x509: certificate signed by unknown authority
```

The workaround described in https://github.com/aquasecurity/trivy/issues/757 is to disable TLS verification for the registry but it would be better if Trivy supported customizing the trusted authorities by being able to provide CA certificate through a ConfigMap for instance (this is what ArgoCD does to support private CA).

**Comment by knqyf263** [comment, marked as answer]
Could you try `SSL_CERT_DIR`?
https://pkg.go.dev/crypto/x509#SystemCertPool

**Comment by gaeljw** [comment]
> Could you try `SSL_CERT_DIR`? https://pkg.go.dev/crypto/x509#SystemCertPool

I don't have much knowledge about Go or Trivy internal, how should I use this variable? Mount a volume with my cert, add the env variable to Trivy and that's it?

I'm using Trivy Operator, so will need to investigate a bit how I could do that.

**Reply by knqyf263** [reply]
Trivy is not relevant. Go will use the environment variable. All you have to do is set the env.
@chen-keinan Can you help?

**Reply by chen-keinan** [reply]
@gaeljw thank you for reporting this issue , currently `trivy-operator` do not support this env `SSL_CERT_DIR ` for scan job.
do you mind opening an issue with [trivy-operator](https://github.com/aquasecurity/trivy-operator) project and we will pick it up from there?

**Reply by gaeljw** [reply]
Sure, here it is: https://github.com/aquasecurity/trivy-operator/issues/1210.
